support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/web/OAuth20HandlerInterceptorAdapter.java

```diff
@@ -48,6 +56,34 @@ public boolean preHandle(final HttpServletRequest request, final HttpServletResp
         return !isAuthorizationRequest(request, response) || requiresAuthenticationAuthorizeInterceptor.preHandle(request, response, handler);
     }
 
+    /**
+    * Is the client requesting is a OAuth "public" client?
+    * An OAuth "public" client is one that does not define a secret like a mobile application.
+    *
+    * @param request the request
+    * @param response the response
+    * @return the boolean
+    */
+    protected boolean clientNeedAuthentication(final HttpServletRequest request, final HttpServletResponse response) {
+        val clientId = OAuth20Utils.getClientIdAndClientSecret(new JEEContext(request, response, sessionStore)).getLeft();
+        val registeredService = OAuth20Utils.getRegisteredOAuthServiceByClientId(servicesManager, clientId);
```

Comment:
If client-id is undefined, return immediately and do not look up the service in registry.

Reply:
Done.